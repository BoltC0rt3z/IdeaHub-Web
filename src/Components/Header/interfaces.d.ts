import { PageHeaderProps } from "antd/lib/page-header";

export interface JoinAndSignInProps {
  joinNow: () => void;
  signIn: () => void;
}

export interface Props extends PageHeaderProps, JoinAndSignInProps {}
