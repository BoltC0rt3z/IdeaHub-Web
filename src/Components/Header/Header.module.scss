@import "../../styles/styles.module.scss";

.header {
  :global(.ant-page-header-heading-title) {
    color: $color-primary;
    font-size: 32px;

    @include respond-to("md-large") {
      font-size: 16px;
    }
  }

  .joinButton {
    :global(.ant-btn) {
      background: red;
    }
  }
}
