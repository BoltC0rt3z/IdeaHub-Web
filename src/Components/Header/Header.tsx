import React from 'react';
import { PageHeader, Row, Col } from 'antd';
import { PageHeaderProps } from 'antd/lib/page-header';
import { ButtonProps } from 'antd/lib/button';
import { Link } from 'react-router-dom';

import { Button } from '../Button';

import styles from './Header.module.scss';

interface Props extends PageHeaderProps {
  hasAccount: boolean;
}

interface BtnProps extends ButtonProps {
  hasAccount: boolean;
}

const HeaderButton: React.FC<BtnProps> = (props) => {
  const { hasAccount, ...restProps } = props;

  return (
    <Row type="flex">
      <Col>
        <Link to={hasAccount ? '/signup' : '/login'}>
          <Button size="large" {...restProps}>
            {hasAccount ? 'Sign Up' : 'Log In'}
          </Button>
        </Link>
      </Col>
    </Row>
  );
};

const Header: React.FC<Props> = (props) => {
  const { hasAccount, ...restProps } = props;

  return <PageHeader className={styles.header} extra={<HeaderButton hasAccount={hasAccount} />} {...restProps} />;
};

export default Header;
