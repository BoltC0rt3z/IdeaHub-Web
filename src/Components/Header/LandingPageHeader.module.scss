@import '../../styles/styles.module.scss';

.header {
  border-bottom: solid 1px $color-light-gray;
  padding: 16px;
  padding-left: 160px;

  @include respond-to('md-large') {
    padding-left: 8px;
  }
}
