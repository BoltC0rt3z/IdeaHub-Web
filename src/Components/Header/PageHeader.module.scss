@import '../../styles/styles.module.scss';

.pageHeader {
  border-bottom: 1px solid $color-light-gray;
  background: $color-light;

  .title {
    cursor: pointer;
  }
}

.avatar {
  .icon {
    background: $color-dark-blue-lighter;
    cursor: pointer;
  }
}
