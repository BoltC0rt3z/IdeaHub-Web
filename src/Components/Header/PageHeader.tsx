import React from 'react';
import { Row, Col, Avatar, Layout } from 'antd';
import { AlignCenterOutlined, UserOutlined } from '@ant-design/icons';

import { Search } from '../../Search';

import styles from './PageHeader.module.scss';
import { DropDown } from '../DropDown';
import { DropDownItem } from '../DropDown/interfaces';

interface Props {
  openSideBar: () => void;
  isMobile: boolean;
  collapsed: boolean;
}

const { Header } = Layout;

const PageHeader: React.FC<Props> = (props) => {
  const { isMobile, collapsed } = props;
  const dropDownItems: DropDownItem[] = [
    {
      name: 'Profile',
      link: '/ideahub/profile',
    },
    {
      name: 'Settings',
      link: '#',
    },
    {
      name: 'Sign out',
      link: '/login',
    },
  ];

  return (
    <Header className={styles.pageHeader}>
      <Row type="flex" justify="space-between">
        <Col>
          <Row type="flex">
            <Col>
              <AlignCenterOutlined style={{ cursor: 'pointer', marginLeft: '-32px' }} onClick={props.openSideBar} />
            </Col>
            <Col>
              <Search />
            </Col>
          </Row>
        </Col>
        <Col className={styles.avatar} style={{ marginRight: !collapsed && !isMobile ? '12rem' : '' }}>
          <DropDown items={dropDownItems}>
            <Avatar icon={<UserOutlined />} className={styles.icon} />
          </DropDown>
        </Col>
      </Row>
    </Header>
  );
};

export default PageHeader;
