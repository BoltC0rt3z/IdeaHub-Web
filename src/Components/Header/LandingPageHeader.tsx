import React from 'react';
import { Row } from 'antd';

import Header from './Header';

import styles from './LandingPageHeader.module.scss';

interface Props {
  hasAccount: boolean;
}

const LandingPageHeader: React.FC<Props> = (props) => {
  const { hasAccount } = props;

  return (
    <Row className={styles.header}>
      <Header title="IdeaHub" hasAccount={hasAccount} />
    </Row>
  );
};

export default LandingPageHeader;
