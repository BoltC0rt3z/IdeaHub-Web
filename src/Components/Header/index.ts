import Header from './Header';
import PageHeader from './PageHeader';
import LandingPageHeader from './LandingPageHeader';

export { Header, PageHeader, LandingPageHeader };
