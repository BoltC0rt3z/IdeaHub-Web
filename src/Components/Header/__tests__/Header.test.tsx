import React from "react";
import { mount, ReactWrapper } from "enzyme";
import { PageHeaderProps } from "antd/lib/page-header";

import Header from "../Header";

describe("header Tests", () => {
  const props: PageHeaderProps = {
    title: "Example Page"
  };

  const wrapper: ReactWrapper = mount(<Header {...props} />);
  it("Should render header without crushing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
