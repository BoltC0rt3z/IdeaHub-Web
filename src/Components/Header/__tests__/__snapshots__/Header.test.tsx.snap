// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header Tests Should render header without crushing 1`] = `ReactWrapper {}`;
